### PR TITLE
Change query for uniswap v2 trades

### LINKED ISSUE #3096: Improve Uniswap trades fetching using the graph
## Problem Definition

Lefteris reported two transactions that were not executed by him but the receiver was one of his accounts 

https://etherscan.io/tx/0xb256ca16c1f0cef96b5fd94c84860473783ed6e1504792f10c4a8bd8113e60bd
https://etherscan.io/tx/0x54cf3ea57f7f23d53a7bd91c3920027131c6320ee089f7409a84f1c88577a4b0

## Logs

image

## Possible solution

Currently when scanning for an account we look in the `from` and `to` field of the swap information since if there are multiple steps in the router the address can appear in both places and some trades where missing before because of this.

A solution would be to see the number of executions made in the transaction and check when there are multiples if the sender is the account we are tracking. This wouldn't imply more queries since all of them should appear when iterating in the results

### PR DESCRIPTION
**Pull request by yabirgb**
In the new subgraph query only for the trades where the address is in the sender

Closes #3096

## What happened

TL;DR: We were querying (from the old query) a extra field that, when using the new graph, was not needed

1. First we used the Uniswap official graph for V2. This graph was not querying correctly the chain and some trades where missing. An update came out but was not pushed
2. We used a different graph up to date with the Uniswap repo but following the query in the old repo we queried for the `to` field. We also query the address in the correct field `from`. So we were looking for the address to be in two fields
3. This caused what was reported in 3096

After looking at the repo for the graph and doing some tests I believe is safe to remove this query where we ask for the address to be in the `from` field.

### CODE REVIEW

**Review by LefterisJP — COMMENTED** [review]
Pushed a fix for incorrect logging string. Extra comma made it into a tuple and resulted in this:

```
2021-06-25T09:24:14.190Z: --- Logging error ---
Traceback (most recent call last):
  File "/home/lefteris/w/rotkehlchen/rotkehlchen/chain/ethereum/graph.py", line 88, in query
    result = self.client.execute(gql(querystr), variable_values=param_values)
  File "/home/lefteris/.virtualenvs/rotkipy37/lib/python3.7/site-packages/gql/client.py", line 76, in execute
    result = self._get_result(document, *args, **kwargs)
  File "/home/lefteris/.virtualenvs/rotkipy37/lib/python3.7/site-packages/gql/client.py", line 84, in _get_result
    return self.transport.execute(document, *args, **kwargs)
  File "/home/lefteris/.virtualenvs/rotkipy37/lib/python3.7/site-packages/gql/transport/requests.py", line 124, in execute
    response.raise_for_status()
  File "/home/lefteris/.virtualenvs/rotkipy37/lib/python3.7/site-packages/requests/models.py", line 943, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 502 Server Error: Bad Gateway for url: https://api.thegraph.com/subgraphs/name/benesjan/uniswap-v2
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/home/lefteris/w/rotkehlchen/rotkehlchen/chain/ethereum/modules/uniswap/uniswap.py", line 795, in _get_trades_graph_for_address
    trades.extend(self._get_trades_graph_v2_for_address(address, start_ts, end_ts))
  File "/home/lefteris/w/rotkehlchen/rotkehlchen/chain/ethereum/modules/uniswap/uniswap.py", line 856, in _get_trades_graph_v2_for_address
    param_values=param_values,
  File "/home/lefteris/w/rotkehlchen/rotkehlchen/chain/ethereum/graph.py", line 108, in query
    raise RemoteError(f'{base_msg}. No retries left.') from e
rotkehlchen.errors.RemoteError: The Graph query to query ($limit: Int!, $offset: Int!, $address: Bytes!, $start_ts: BigInt!, $end_ts: BigInt!){swaps ( first: $limit, skip: $offset, where: { from: $address, timestamp_gte: $start_ts, timestamp_lte: $end_ts, } ) { transaction { swaps { id logIndex sender to timestamp pair { token0 { id decimals name symbol } token1 { id decimals name symbol } } amount0In amount0Out amount1In amount1Out } } }} failed due to 502 Server Error: Bad Gateway for url: https://api.thegraph.com/subgraphs/name/benesjan/uniswap-v2. No retries left.
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/usr/lib/python3.7/logging/handlers.py", line 69, in emit
    if self.shouldRollover(record):
  File "/usr/lib/python3.7/logging/handlers.py", line 185, in shouldRollover
    msg = "%s\n" % self.format(record)
  File "/usr/lib/python3.7/logging/__init__.py", line 869, in format
    return fmt.format(record)
  File "/usr/lib/python3.7/logging/__init__.py", line 608, in format
    record.message = record.getMessage()
  File "/usr/lib/python3.7/logging/__init__.py", line 369, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "/home/lefteris/w/rotkehlchen/rotkehlchen/api/rest.py", line 270, in _do_query_async
    result = getattr(self, command)(**kwargs)
  File "/home/lefteris/w/rotkehlchen/rotkehlchen/api/rest.py", line 2208, in _eth_module_query
    result = getattr(module_obj, method)(**kwargs)
  File "/home/lefteris/w/rotkehlchen/rotkehlchen/chain/ethereum/modules/uniswap/uniswap.py", line 1280, in get_trades_history
    only_cache=False,
  File "/home/lefteris/w/rotkehlchen/rotkehlchen/chain/ethereum/modules/uniswap/uniswap.py", line 710, in _get_trades
    end_ts=to_timestamp,
  File "/home/lefteris/w/rotkehlchen/rotkehlchen/chain/ethereum/modules/uniswap/uniswap.py", line 781, in _get_trades_graph
    trades = self._get_trades_graph_for_address(address, start_ts, end_ts)
  File "/home/lefteris/w/rotkehlchen/rotkehlchen/chain/ethereum/modules/uniswap/uniswap.py", line 799, in _get_trades_graph_for_address
    f'between {start_ts} and {end_ts}. {str(e)}',
Message: 'Error querying uniswap v2 trades using graph for address 0xd1b8d347Fd50dC7838A8cEB4294B6621B0b300f6 '
Arguments: ('between 0 and 1624612814. The Graph query to query ($limit: Int!, $offset: Int!, $address: Bytes!, $start_ts: BigInt!, $end_ts: BigInt!){swaps ( first: $limit, skip: $offset, where: { from: $address, timestamp_gte: $start_ts, timestamp_lte: $end_ts, } ) { transaction { swaps { id logIndex sender to timestamp pair { token0 { id decimals name symbol } token1 { id decimals name symbol } } amount0In amount0Out amount1In amount1Out } } }} failed due to 502 Server Error: Bad Gateway for url: https://api.thegraph.com/subgraphs/name/benesjan/uniswap-v2. No retries left.',)
```

**Review by LefterisJP — APPROVED** [review]
LGTM. Seems to work and indeed skip those 2 trades in my account which were not done by me.

Now the only thing is that we should remember to write in the release notes a note that says that if the users have any uniswap trades that they do not recognize they should delete the uniswap data and repull it.

**Review by LefterisJP — COMMENTED** [review]
Hmm seems there is one actual failing test here. I will investigate.

`python pytestgeventwrapper.py -xs  --pdb rotkehlchen/tests/api/test_uniswap.py::test_get_uniswap_exotic_history --pdb `
